onnxruntime/python/tools/transformers/optimizer.py

```diff
@@ -251,6 +257,7 @@ def _get_optimization_options(args):
 def optimize_model(input,
                    model_type='bert',
                    num_heads=12,
+                   head_size=0,
```

Comment:
it's better to infer the head_size from num_heads and hidden_size to avoid compatibility issue

Reply:
As I mentioned above, (self.hidden_size != self.num_heads * self.head_size) in case head pruned architecture. The default value (0) indicates that the heads are not pruned and it infers the `head_size` based on the `hidden_size` and `num_heads`.